src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -29,104 +29,123 @@ class ThetaStarPathPlanner : public PathPlanner
                   const std::vector<Obstacle> &obstacles) override;
 
    private:
-    typedef std::pair<int, int> CellCoordinate;
-    typedef std::pair<double, CellCoordinate> OpenListCell;
+    class Coordinate : public std::pair<int, int>
+    {
+       public:
+        Coordinate(int row, int col) : std::pair<int, int>(row, col) {}
+
+        Coordinate() : std::pair<int, int>() {}
+
+        int row(void)
+        {
+            return this->first;
+        }
+
+        int col(void)
+        {
+            return this->second;
+        }
+    };
 
-    class GridCell
+    class CellHeuristic
     {
        public:
-        GridCell(int parent_i, int parent_j, double f, double g, double h)
-            : parent_i_(parent_i), parent_j_(parent_j), f_(f), g_(g), h_(h)
+        CellHeuristic(Coordinate parent, double f, double g, double h)
+            : parent(parent), f(f), g(g), h(h)
         {
         }
 
-        int parent_i_, parent_j_;
-        double f_, g_, h_;
+        Coordinate parent;
+        double f, g, h;
     };
 
     /**
      * Returns if a cell is within bounds of grid
-     * @param row y position of cell
-     * @param col x position of cell
+     *
+     * @param test_coord Coordinate to consider
      *
      * @return true if cell is valid
      */
-    bool isValid(int row, int col);
+    bool isCoordValid(Coordinate test_coord);
 
     /**
      * Returns if a cell is unblocked
-     * @param row y position of cell
-     * @param col x position of cell
+     *
+     * @param test_coord Coordinate to consider
      *
      * @return true if cell is unblocked
      */
-    bool isUnBlocked(int row, int col);
+    bool isUnBlocked(Coordinate test_coord);
 
     /**
      * Returns if a cell is the destination
-     * @param row y position of cell
-     * @param col x position of cell
+     *
+     * @param test_coord Coordinate to consider
      * @param dest destination cell
      *
      * @return true if cell is the destination
      */
-    bool isDestination(int row, int col, CellCoordinate dest);
+    bool isDestination(Coordinate test_coord, Coordinate dest);
```

Comment:
The implementation of this function is just checking equality. Can we either change the function name to `isEqual` or just implement the equality operator for the `Coordinate` class? I was really confused at first why 2 arguments were needed to check if something was the destination.

Reply:
can just use == here